import os

# Создаем основную директорию для программы
program_path = r"C:\Users\Yaroslav\Desktop\UAB"
os.makedirs(program_path, exist_ok=True)

# Создаем директорию для интерфейса программы
interface_path = os.path.join(program_path, "interface")
os.makedirs(interface_path, exist_ok=True)

# Создаем основной Python скрипт
main_script_path = os.path.join(program_path, "unity_project_creator.py")
with open(main_script_path, 'w') as file:
    file.write("# Основной скрипт программы")

# Создаем файл интерфейса
ui_file_path = os.path.join(interface_path, "interface.ui")
with open(ui_file_path, 'w') as file:
    file.write("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
               "<ui version=\"4.0\">\n"
               " <class>MainWindow</class>\n"
               " <widget class=\"QMainWindow\" name=\"MainWindow\">\n"
               "  <property name=\"geometry\">\n"
               "   <rect>\n"
               "    <x>0</x>\n"
               "    <y>0</y>\n"
               "    <width>600</width>\n"
               "    <height>400</height>\n"
               "   </rect>\n"
               "  </property>\n"
               "  <property name=\"windowTitle\">\n"
               "   <string>UAB</string>\n"
               "  </property>\n"
               "  <widget class=\"QWidget\" name=\"centralwidget\">\n"
               "   <widget class=\"QPushButton\" name=\"create_project_button\">\n"
               "    <property name=\"geometry\">\n"
               "     <rect>\n"
               "      <x>20</x>\n"
               "      <y>300</y>\n"
               "      <width>561</width>\n"
               "      <height>32</height>\n"
               "     </rect>\n"
               "    </property>\n"
               "    <property name=\"text\">\n"
               "     <string>Create Project</string>\n"
               "    </property>\n"
               "   </widget>\n"
               "   <widget class=\"QLineEdit\" name=\"project_name\">\n"
               "    <property name=\"geometry\">\n"
               "     <rect>\n"
               "      <x>20</x>\n"
               "      <y>30</y>\n"
               "      <width>561</width>\n"
               "      <height>32</height>\n"
               "     </rect>\n"
               "    </property>\n"
               "    <property name=\"placeholderText\">\n"
               "     <string>Project Name</string>\n"
               "    </property>\n"
               "   </widget>\n"
               "   <widget class=\"QLineEdit\" name=\"project_path\">\n"
               "    <property name=\"geometry\">\n"
               "     <rect>\n"
               "      <x>20</x>\n"
               "      <y>80</y>\n"
               "      <width>561</width>\n"
               "      <height>32</height>\n"
               "     </rect>\n"
               "    </property>\n"
               "    <property name=\"placeholderText\">\n"
               "     <string>Project Path</string>\n"
               "    </property>\n"
               "   </widget>\n"
               "   <widget class=\"QPushButton\" name=\"browse_button\">\n"
               "    <property name=\"geometry\">\n"
               "     <rect>\n"
               "      <x>510</x>\n"
               "      <y>130</y>\n"
               "      <width>75</width>\n"
               "      <height>32</height>\n"
               "     </rect>\n"
               "    </property>\n"
               "    <property name=\"text\">\n"
               "     <string>Browse</string>\n"
               "    </property>\n"
               "   </widget>\n"
               "   <widget class=\"QTextEdit\" name=\"description\">\n"
               "    <property name=\"geometry\">\n"
               "     <rect>\n"
               "      <x>20</x>\n"
               "      <y>180</y>\n"
               "      <width>561</width>\n"
               "      <height>100</height>\n"
               "     </rect>\n"
               "    </property>\n"
               "    <property name=\"placeholderText\">\n"
               "     <string>Project Description</string>\n"
               "    </property>\n"
               "   </widget>\n"
               "   <widget class=\"QLabel\" name=\"status\">\n"
               "    <property name=\"geometry\">\n"
               "     <rect>\n"
               "      <x>20</x>\n"
               "      <y>340</y>\n"
               "      <width>561</width>\n"
               "      <height>32</height>\n"
               "     </rect>\n"
               "    </property>\n"
               "    <property name=\"text\">\n"
               "     <string>Status: Ready</string>\n"
               "    </property>\n"
               "   </widget>\n"
               "  </widget>\n"
               " </widget>\n"
               " <resources/>\n"
               " <connections/>\n"
               "</ui>")